### PR TITLE
Initial commit of the Prometheus exporter

### PR DESCRIPTION
**Pull request by rakyll**
odeke-em@ and I found a way to reuse the Prometheus Go package
to render the endpoint. This change is an initial commit with the APIs.

The exporter is heavily under construction.

Updates #168.

### CODE REVIEW

**Comment by rakyll** [comment]
An example of the export endpoint:

export

File: exporter/stats/prometheus/prometheus.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	if e.opts.OnError != nil {
+		e.opts.OnError(err)
+		return
+	}
+	log.Printf("Failed to export to Prometheus: %v", err)
+}
+
+// Export exports to the Prometheus if view data has one or more rows.
+func (e *Exporter) Export(vd *stats.ViewData) {
+	if len(vd.Rows) > 0 {
```
Comment (odeke-em) [inline exporter/stats/prometheus/prometheus.go]: I think we meant the opposite :)

File: exporter/stats/prometheus/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		for _, t := range r.Tags {
+			labels = append(labels, internal.Sanitize(t.Key.Name()))
+		}
+		// Sanitize the view name.
+		desc := prometheus.NewDesc(view.Name(), view.Description(), labels, nil)
+		e.c.descCh <- desc
+
+		// TODO(jbd): Support other metric types.
+		switch v := r.Data.(type) {
+		case *stats.CountData:
```
Comment (odeke-em) [inline exporter/stats/prometheus/prometheus.go]: This is awesome. PTAL at the one for DistributionAggregation

https://github.com/orijtech/opencensus-go/blob/d6201c4289f86c30fa85ba0e103f9ad1e201988d/exporter/stats/prometheus/prometheus.go#L188-L207

File: exporter/stats/prometheus/prometheus.go (hunk 1 above)
Comment (rakyll) [reply exporter/stats/prometheus/prometheus.go]: Oops, sorry. Fixed.

File: exporter/stats/prometheus/prometheus.go (hunk 2 above)
Comment (rakyll) [reply exporter/stats/prometheus/prometheus.go]: Neat 👍

File: examples/stats/prometheus/main.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	// Set reporting period to report data at every second.
+	stats.SetReportingPeriod(1 * time.Second)
+
+	// Subscribe will allow view data to be exported.
+	// Once no longer need, you can unsubscribe from the view.
+	if err := view.Subscribe(); err != nil {
+		log.Fatalf("Cannot subscribe to the view: %v", err)
+	}
+
+	// Record data points.
```
Comment (odeke-em) [inline examples/stats/prometheus/main.go]: One other thing, could we get this to continually send/simulate data
```go
// Simulate transcoding going on in the background
	go func() {
		ctx := context.Background()
		for {
			// Record data points.
			stats.Record(ctx, videoSize.M(rand.Int63()))
			<-time.After(time.Millisecond * time.Duration(1+rand.Intn(450)))
		}
	}()
```
which will demo the whole time series concept and on refresh will produce new data.

**Pull request merged**